src/systems/systems/src/air_conditioning/mod.rs

```diff
@@ -0,0 +1,387 @@
+use self::acs_controller::ACSController;
+
+use crate::{
+    overhead::{OnOffFaultPushButton, ValueKnob},
+    shared::{CabinAltitude, EngineCorrectedN1, LgciuWeightOnWheels},
+    simulation::{
+        InitContext, SimulationElement, SimulationElementVisitor, SimulatorWriter, UpdateContext,
+        VariableIdentifier, Write,
+    },
+};
+
+use std::{collections::HashMap, time::Duration};
+
+use uom::si::{
+    f64::*, mass_rate::kilogram_per_second, pressure::hectopascal,
+    thermodynamic_temperature::degree_celsius,
+};
+
+pub mod acs_controller;
+pub mod cabin_air;
+
+pub trait DuctTemperature {
+    fn duct_demand_temperature(&self) -> HashMap<&'static str, ThermodynamicTemperature>;
+}
+
+pub trait FlowControlValveSignal {
+    fn should_open_fcv(&self) -> [bool; 2];
+}
+
+pub trait PackFlow {
+    fn pack_flow(&self) -> MassRate;
+}
+
+pub struct AirConditioningSystem {
+    acs_overhead: AirConditioningSystemOverhead,
+    acsc: ACSController,
+    pack_flow_valve: [PackFlowValve; 2],
+    // TODO: pack: [AirConditioningPack; 2],
+    // TODO: mixer_unit: MixerUnit,
+    // TODO: trim_air_system: TrimAirSystem,
+}
+
+impl AirConditioningSystem {
+    pub fn new(context: &mut InitContext, cabin_zone_ids: Vec<&'static str>) -> Self {
+        Self {
+            acs_overhead: AirConditioningSystemOverhead::new(context, &cabin_zone_ids),
+            acsc: ACSController::new(context, cabin_zone_ids),
+            pack_flow_valve: [
+                PackFlowValve::new(context, 1),
+                PackFlowValve::new(context, 2),
+            ],
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        engines: [&impl EngineCorrectedN1; 2],
+        pressurization: &impl CabinAltitude,
+        lgciu: [&impl LgciuWeightOnWheels; 2],
+    ) {
+        self.acsc.update(
+            context,
+            &self.acs_overhead,
+            &self.pack_flow_valve,
+            engines,
+            pressurization,
+            lgciu,
+        );
+
+        for fcv in self.pack_flow_valve.iter_mut() {
```

Comment:
That is me mixing naming conventions in my head! `fcv` stands for Flow Control Valve, which is the same thing as the `pack_flow_valve`. Changed now to `pack_fv` to avoid confusion.